test/test-slowparse.js

```diff
@@ -597,6 +597,12 @@ module.exports = function(Slowparse, window, document, validators) {
     ok(!result.error, "no error on omitted </p>");
   });
 
+  test("parsing elements with optional close tags: <datalist><option>a<option>b</datalist>", function() {
+    var html = '<datalist><option>a<option>b</datalist>';
+    var result = parse(html);
+    ok(!result.error, "no error on omitted </option>");
+  });
+
```

Comment:
given the additional entries you added, it's important to add test cases for those, too.